app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/VoiceAnalysisService.kt

```diff
@@ -73,19 +74,21 @@ class VoiceAnalysisService : IntentService("VoiceAnalysisService") {
         aacRecorder =
             AacRecorder(voiceAnalysisOption)
         aacRecorder?.startRecording()
-            ?.subscribeOn(Schedulers.computation())
+            ?.subscribeOn(Schedulers.io())
```

Comment:
Wouldn't it be better to just use .apply { } on AacRecorder and be sure that all subscriptions will be started as expected and no unnecessary null checks will be needed?

Reply:
also on the second note aac recorder seems to me much more like lateinit var than nullable 🤔 is that possible that other methods will be invoked before it's set?

Reply:
the recorder is being nullified when webRTC stream starts to release the audio recording resources. It is being reinitialisied after the stream.